Projects/UI/HGDesignSystem/Sources/Extensions/View+.swift

```diff
@@ -8,6 +8,61 @@
 import SwiftUI
 
 public extension View {
+    /// Modifier 분기 적용
+    @ViewBuilder
+    func `if`<Content: View>(_ condition: Bool, content: (Self) -> Content) -> some View {
+        if condition {
+            content(self)
+        } else {
+            self
+        }
+    }
     
+    /// Radius 설정
+    func setRadius(_ radius: CGFloat, corners: [CornerType] = CornerType.allCases) -> some View {
+        self.clipShape(
+            .rect(
+                topLeadingRadius: corners.contains(.topLeft) ? radius : .zero,
+                bottomLeadingRadius: corners.contains(.bottomLeft) ? radius : .zero,
+                bottomTrailingRadius: corners.contains(.bottomRight) ? radius : .zero,
+                topTrailingRadius: corners.contains(.topRight) ? radius : .zero
+            )
+        )
+    }
+    
+    /// Set rounded border
+    @ViewBuilder func roundedBorder(
+        _ color: Color,
+        radius: CGFloat = 8,
+        linewidth: CGFloat = 1
+    ) -> some View {
+        self
+            .clipShape(RoundedRectangle(cornerRadius: radius))
+            .overlay {
+                RoundedRectangle(cornerRadius: radius)
+                    .stroke(color, lineWidth: linewidth)
+            }
+    }
+    
+    /// Horizontal infinity width
+    func hSpacing(_ alignment: Alignment = .leading) -> some View {
+        self.frame(maxWidth: .infinity, alignment: alignment)
+    }
+    
+    /// 기본 프로그래스 로딩뷰
+    @ViewBuilder func isLoading(_ state: Bool) -> some View {
+        self
+            .disabled(state)
+            .overlay {
+                if state {
+                    Color.black.opacity(0.25)
+                        .ignoresSafeArea()
+                        .overlay {
+                            ProgressView() // TODO: 추후 디자인에 맞게 변경
+                                .progressViewStyle(.circular)
+                        }
+                }
+            }
```

Comment:
몇가지 끄적여봅니당
1. 탭바 화면까지 다가릴수잇는지 -> 네비영역, 탭바영역이 색이 어색해질거같아서
아마 디자인에따라서 이것도 달라질예정일거 같긴해보여
2. extension정의한 if 함수 이용
```suggestion
      self.disabled(state)
            .if(state) {
                $0.overlay {
                    Color.black.opacity(0.25)
                        .ignoresSafeArea()
                        .overlay {
                            ProgressView() // TODO: 추후 디자인에 맞게 변경
                                .progressViewStyle(.circular)
                        }
                }
            }
```

Reply:
1. 최상단에서 사용하면 ignoreSafeArea() 있어서 잘 커버해주지 않을까유!? 테스트 해보겠슴당
2. 전자가 더 가독성 있지 않아!?

Reply:
1. 뷰자체가 네비와 탭뷰 하위로들어가버려서 그상위뷰까지 가리는게 안됫던 기억이있어서!!
2. if를 extension으로 제공해놔서 
지금 제공해놓은 if가 `if state` 처리같은거 쓸때 대신 사용하는거라고 이해해서
코멘트 달아봐씀 !!
현코드 유지해도 상관없어용

Reply:
이건 나중에 루트뷰에서 environment로 isLoadingState 넣어주고 루트뷰에 오버레이 하는식으로 해보자 일단 로딩 우리 디자인에 없으니 패스 어때1?